### ISSUE TITLE
Training loss curve for jigsaw task

### ISSUE THREAD

**Issue by prigoyal** [MehdiNoroozi/JigsawPuzzleSolver#6]
Hi,

I am wondering if I can get the training loss curve for jigsaw training? I also tried running the code directly but it segfaults after 2000 iterations with trace below. I would appreciate if you can point to how to resolve this particular error if the loss curve is not available. Thanks for your help. 

```
0x00007ffff7813760 in caffe::DataTransformer<float>::Transform(caffe::Datum&, float*) () from /private/home/prigoyal/local/JigsawPuzzleSolver/caffe-master-jps++/build/lib/libcaffe.so
(gdb) bt
#0  0x00007ffff7813760 in caffe::DataTransformer<float>::Transform(caffe::Datum&, float*) () from /private/home/prigoyal/local/JigsawPuzzleSolver/caffe-master-jps++/build/lib/libcaffe.so
#1  0x00007ffff7814ad6 in caffe::DataTransformer<float>::Transform(caffe::Datum&, caffe::Blob<float>*) () from /private/home/prigoyal/local/JigsawPuzzleSolver/caffe-master-jps++/build/lib/libcaffe.so
#2  0x00007ffff7889624 in caffe::DataLayer<float>::load_batch(caffe::Batch<float>*) () from /private/home/prigoyal/local/JigsawPuzzleSolver/caffe-master-jps++/build/lib/libcaffe.so
#3  0x00007ffff784925b in caffe::BasePrefetchingDataLayer<float>::InternalThreadEntry() () from /private/home/prigoyal/local/JigsawPuzzleSolver/caffe-master-jps++/build/lib/libcaffe.so
#4  0x00007ffff781e775 in caffe::InternalThread::entry(int, caffe::Caffe::Brew, int, int, bool) () from /private/home/prigoyal/local/JigsawPuzzleSolver/caffe-master-jps++/build/lib/libcaffe.so
#5  0x00007ffff62af5d5 in ?? () from /usr/lib/x86_64-linux-gnu/libboost_thread.so.1.58.0
#6  0x00007ffff60886ba in start_thread (arg=0x7fffc1d11700) at pthread_create.c:333
#7  0x00007ffff65cb41d in clone () at ../sysdeps/unix/sysv/linux/x86_64/clone.S:109
```

cc @imisra

**Comment by prigoyal** [comment]
@MehdiNoroozi , we sent you an email around July 10 asking for loss curve and would appreciate any response either on email or this issue. Thanks a lot :)
